rs/config/src/execution_environment.rs

```diff
@@ -317,6 +327,8 @@ impl Default for Config {
             max_canister_memory_size: NumBytes::new(
                 MAX_STABLE_MEMORY_IN_BYTES + MAX_WASM_MEMORY_IN_BYTES,
             ),
+            subnet_callback_soft_cap: 1_000_000,
+            canister_callback_quota: 50,
```

Comment:
Is there a basis for these numbers? So IIUC the hard cap is `canisters_count * canister_callback_quota` from the comments above. This can be larger than the soft subnet wide quota from 1_000_000 / 50 = 20_000 canisters (i.e. higher for > 20k). So how many canisters can we conceivably have on a subnet?

Come to think of it, isn't this also in conflict with the 500 messages limit in output queues?